Initial stub

dadlex opened the pull request.

dadlex committed:
Create initial stub

dadlex committed:
Create initial stub

dadlex committed:
Merge remote-tracking branch 'remotes/origin/master' into feature-initial-stub

# Conflicts:
#	README.md

veronikaKochugova commented on ci/.gitlab-ci.yml:
```diff
@@ -0,0 +1,204 @@
+variables:
+  DOCKER_HOST: tcp://docker:2375
+  DOCKER_DRIVER: overlay2
+  IMAGE_NAME: emcmongoose/mongoose-storage-driver-pravega
```
change image name

dadlex replied:
yeah, sure, CI is just copied from pravega though. so don't expect much there. we still need to figure out gitlab mirroring and whatnot

veronikaKochugova commented on ci/.gitlab-ci.yml:
```diff
@@ -0,0 +1,204 @@
+  DOCKER_HOST: tcp://docker:2375
+  DOCKER_DRIVER: overlay2
+  IMAGE_NAME: emcmongoose/mongoose-storage-driver-pravega
+  IMAGE_FILE_NAME: build/mongoose-storage-driver-pravega.tar
```
change file name

veronikaKochugova commented on ci/docker/Dockerfile:
```diff
@@ -0,0 +1,23 @@
+ARG VERSION
+RUN mkdir -p $HOME/.mongoose/${BASE_VERSION}/ext
+COPY --from=0 /build/libs/mongoose-storage-driver-pravega-kvs-${VERSION}.jar /tmp/
+ADD ci/docker/entrypoint_storage_driver_pravega_kvs.sh /opt/mongoose/entrypoint_storage_driver_pravega_kvs.sh
```
i see only entrypoint_storage_drivet_pravega.sh

dadlex replied:
you're right. missed that. only checked the jar mode works so far

veronikaKochugova commented on src/main/resources/config-schema-storage-pravega-kvs.yaml:
```diff
@@ -0,0 +1,30 @@
+    control:
+      scope: boolean
+      timeoutMillis: long
+    event:
```
will we change "event" on smth else?

dadlex replied:
Sure. I guess we can do that already. storage-driver-hashing-key-enabled sounds good enough to me. you?

veronikaKochugova commented on src/main/java/com/emc/mongoose/storage/driver/pravega/kvs/RoutingKeyFunction.java:
```diff
@@ -0,0 +1,10 @@
+import com.emc.mongoose.base.item.DataItem;
+import java.util.function.Function;
+
+public interface RoutingKeyFunction<I extends DataItem>
```
Do we need to use Hashing Key instead of Routing Key? This can be confusing.

dadlex replied:
I left it mainly to remember that we may be able to reuse that code when we het to key families. I'm not sure we can use these classes directly, so I'd prefer they stay rountingKey until we are sure they are working for KVT

dadlex replied:
The name should change, that's without doubt

dadlex committed:
Correct entrypoint to docker image
Correct image name for gitlab ci
Change parameter name event-key to hashing key. earlier represented routing key

AndreyKoltsov1997 commented on src/test/java/com/emc/mongoose/storage/driver/pravega/kvs/util/docker/PravegaNodeContainer.java:
```diff
@@ -0,0 +1,59 @@
+
+	public static final int PORT = 9090;
+	private static final Logger LOG = Logger.getLogger(PravegaNodeContainer.class.getSimpleName());
+	private static final String IMAGE_NAME = "pravega/pravega:0.3.2";
```
If the test has been moved from Storage Driver Pravega, I assume the image should be changed in each of the drivers since `pravega/pravega:0.3.2` is 2 years old now.

dadlex replied:
Sure, this test isn't currently used though,

AndreyKoltsov1997 commented on README.md:
```diff
@@ -1,6 +1,211 @@
+[![CI status](https://travis-ci.org/emc-mongoose/mongoose-storage-driver-pravega.svg?branch=master)](https://travis-ci.org/emc-mongoose/mongoose-storage-driver-pravega/builds)
+[![Maven metadata URL](https://img.shields.io/maven-metadata/v/http/central.maven.org/maven2/com/github/emc-mongoose/mongoose-storage-driver-pravega/maven-metadata.xml.svg)](http://central.maven.org/maven2/com/github/emc-mongoose/mongoose-storage-driver-pravega-kvs)
+[![Docker Pulls](https://img.shields.io/docker/pulls/emcmongoose/mongoose-storage-driver-pravega.svg)](https://hub.docker.com/r/emcmongoose/mongoose-storage-driver-pravega/)
 
```
I see a lot of references to `mongoose-storage-driver-pravega` (CI status, maven metadata, docker pulls). I'd suggest to either change them onto the appropriate `mongoose-srorage-driver-pravega-kvs` links or remove those leaving a "TODO" so it won't confuse anyone.

dadlex replied:
Added a task to see if links get updated when driver is first released

dadlex committed:
Fix links
Delete CI link so far
